activities.el

```diff
@@ -801,6 +851,116 @@ activity's name is NAME."
                   "In the meantime, it's recommended to not use buffers of this major mode in an activity's layout; or you may simply ignore this error and use the other buffers in the activity.")
           (current-buffer)))))
 
+(defvar activities--age-spec
+  `((?Y "year"   "years"   ,(round (* 60 60 24 365.2425)))
+    (?M "month"  "months"  ,(round (* 60 60 24 30.436875)))
+    (?w "week"   "weeks"   ,(* 60 60 24 7))
+    (?d "day"    "days"    ,(* 60 60 24))
+    (?h "hour"   "hours"   ,(* 60 60))
+    (?m "minute" "minutes" 60)
+    (?s "second" "seconds" 1))
+  "Age specification.  See `magit--age-spec', which this duplicates.")
+
+(defun activities--age (age &optional abbreviate)
+  "Summarize AGE and possibly ABBREVIATE.
+Adapted from `magit--age'."
+  (cl-labels ((fn (age spec)
+                (pcase-let ((`(,char ,unit ,units ,weight) (car spec)))
+                  (let ((cnt (round (/ age weight 1.0))))
+                    (if (or (not (cdr spec))
+                            (>= (/ age weight) 1))
+                        (list cnt (cond (abbreviate char)
+                                        ((= cnt 1) unit)
+                                        (t units)))
+                      (fn age (cdr spec)))))))
+    (fn age activities--age-spec)))
+
+(defun activities--oldest-age (activities)
+  "Return the age in seconds of the oldest activity in ACTIVITIES."
+  (cl-loop for (_name . act) in activities maximize
+	   (cl-loop
+	    for type in '(default last)
+ 	    for state = (cl-struct-slot-value 'activities-activity type act)
+	    if state
+	    for etc = (activities-activity-state-etc state)
+	    maximize (float-time (time-since (map-elt etc 'time))))))
+
+(defun activities--annotate (max-age oldest-possible)
```

Comment:
Forgive my laziness, since my time is limited, but could this be done without defining this top-level symbol?  I really like to try to avoid defining new ones.

Reply:
I could shove this into `activities-completing-read` just as I did the completion table, but it's a pretty meaty function.  What do you think?

Reply:
I managed to squeeze the annotation function in there.